x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Arithmetics.java

```diff
@@ -33,20 +39,12 @@ static Number add(Number l, Number r) {
         return Integer.valueOf(Math.addExact(l.intValue(), r.intValue()));
     }
 
-    static ZonedDateTime add(ZonedDateTime l, Period r) {
-        if (l == null || r == null) {
-            return null;
-        }
-
-        return l.plus(r);
+    static Temporal add(Temporal l, Period r) {
```

Comment:
Why delegate to `periodArithmetics` instead of having its content here?
Same applies to the other add and sub methods.

Reply:
Leftover of some intermediate state. will change.

Reply:
Ah, sorry no, this is because the periodArithmetics/durationArithmetics are also used for the `minus`.

Reply:
The type checks have been added to `BinaryArithmeticOperation`

Reply:
But how can I have one method since There is some difference on how Duration & Period are handled, I wouldn't like to make instance of on the `TemporalAmount`.